src/main/scala/org/constellation/primitives/EdgeDAO.scala

```diff
@@ -107,53 +115,100 @@ class ThreadSafeTipService() {
 
     if (dao.nodeState == NodeState.Ready && acceptedCBSinceSnapshot.nonEmpty) {
 
-      val nextSnapshot = Snapshot(snapshot.hash, acceptedCBSinceSnapshot.sorted)
+      val minTipHeight = getMinTipHeight()
+      dao.metricsManager ! UpdateMetric("minTipHeight", minTipHeight.toString)
+
+      val nextHeightInterval = lastSnapshotHeight + dao.processingConfig.snapshotHeightInterval
+
+      val canSnapshot = minTipHeight > (nextHeightInterval + dao.processingConfig.snapshotHeightDelayInterval)
+      if (!canSnapshot) {
+        dao.metricsManager ! IncrementMetric("snapshotHeightIntervalConditionNotMet")
+      } else {
+
+        val maybeDatas = acceptedCBSinceSnapshot.map(dao.checkpointService.get)
 
-      // TODO: Make this a future and have it not break the unit test
-      // Also make the db puts blocking, may help for different issue
-      if (snapshot != Snapshot.snapshotZero) {
-        dao.metricsManager ! IncrementMetric("snapshotCount")
+        val blocksWithinHeightInterval = maybeDatas.filter {
+          _.exists(_.height.exists { h =>
+            h.min > lastSnapshotHeight && h.min <= nextHeightInterval
+          })
+        }
+
+        if (blocksWithinHeightInterval.isEmpty) {
+          dao.metricsManager ! IncrementMetric("snapshotNoBlocksWithinHeightInterval")
+        } else {
 
-        // Write snapshot to file
-        tryWithMetric({
-          val maybeBlocks = snapshot.checkpointBlocks.map {h =>
-            val res = checkpointCache.get(h)
-            if (res.isEmpty) dao.dbActor.getCheckpointCacheData(h).map{_.checkpointBlock}
-            else res
+          val blockCaches = blocksWithinHeightInterval.map {
```

Comment:
this whole snapshot flow is long enough it seems like it might be a good candidate for a method?

Reply:
Yes we should also pull it into it's own class